### ISSUE TITLE
Add newlines for created MarkdownString

### ISSUE THREAD

**Issue by dbaeumer** [Microsoft/vscode-languageserver-node#328]
See code at:

https://github.com/Microsoft/vscode-languageserver-node/blob/master/client/src/protocolConverter.ts#L187

**Comment by Gama11** [comment]
Sorry if this is a bit off-topic, but is using `MarkedString[]` the only way to achieve this formatting with a thin horizontal separator in a language server <sup>[1]</sup>? It doesn't appear like there's a way to do this with `MarkupContent`, yet `MarkedString` is [deprectated in favor of it](https://github.com/Microsoft/vscode-languageserver-node/blob/7994319e90866970f68fdd828b4dc03da5fdc423/types/src/main.ts#L1057)... So is this formatting effectively deprecated as well?

<sup>[1]</sup> Of course there's markdown separators as well, but the way they're rendered in VSCode is quite different.

**Comment by dbaeumer** [comment]
Good point. The only way how I see to let the client know about this is to allow MarkupContent[] in a Hover. But this needs to be spec in the LSP correctly.

Can you open an issue against the LSP repository and we will see if this needs to be added. You can always use the deprecated way for now.

**Comment by Gama11** [comment]
Yeah, that's what I've done for now. Created an issue here: https://github.com/Microsoft/language-server-protocol/issues/518
